docker/compile/Dockerfile.ubuntu20.04

```diff
@@ -31,12 +31,13 @@ RUN apt-get update && apt-get -y install \
       libunwind8 \
       libcppunit-dev \
       patch \
+      python3 \
       python3-dev \
-      python \
+      python3-pip \
```

Comment:
is this needed for something? pip2/pip3 wasn't included before

Reply:
It might not be needed, but I ran into an issue when trying to `pip3` without it. This initial commit was just a placeholder really. I'm planning to go through each image type. Changes from `virtualenv` to `venv` will probably be a bigger change in another commit. Unless you want to just collaborate on that kind of change here. We can do something similar to your removal of `virtualenv` in #3583 .